lib/grape/validations/validators/length_validator.rb

```diff
@@ -18,6 +19,12 @@ def initialize(attrs, options, required, scope, **opts)
         def validate_param!(attr_name, params)
           param = params[attr_name]
 
+          if param.nil?
+            return if @allow_nil
```

Comment:
I was looking into the code to understand how `allow_blank` interops with other validators and saw that it's an attribute in the [base class](https://github.com/ruby-grape/grape/blob/master/lib/grape/validations/validators/base.rb#L24).

So should we just use this option instead of introducing `allow_nil` ?

Reply:
That was my thinking. 🤷‍♂️

Reply:
I can help decide: does `allow_nil` actually behave like `allow_blank` in this implementation? Meaning is `nil` vs. `[]` treated the same?

Reply:
technically no since `allow_nil` is for only `nil` values while `allow_blank` would mean ignoring both `nil` and `empty` values. Not sure if we want to use `allow_blank` or `allow_nil`. 

I personally do prefer more specific constraints thus `allow_nil` but I am also okay if we decide to ignore `blank` values if the `allow_blank` flag is set to true.

Reply:
@OuYangJinTing write specs for these scenarios and let's decide?

Again, thanks so much for being patient. API changes are one-way, we can't easily undo them, so it's worth spending the time.

Reply:
@dhruvCW Introducing `allow_nil` here only will make it potentially different from other parameter types. Without looking at specs and current behavior I would say `allow_blank` for arrays/lists should behave like `blank?` and if we wanted a `allow_nil` then we should probably implement it for all other parameter types next to `allow_blank`.

Reply:
> @OuYangJinTing write specs for these scenarios and let's decide?
> 
> Again, thanks so much for being patient. API changes are one-way, we can't easily undo them, so it's worth spending the time.

Sorry, I'm a little confused.
Do you need to roll back to commit [171407aeef935ab918335c5ea324ab18a96a19d4](https://github.com/ruby-grape/grape/commit/171407aeef935ab918335c5ea324ab18a96a19d4) and provide a more detailed explanation?

Reply:
Let's wait for @ericproulx to help us out with a strong opinion

Reply:
Hmm, I'll make the changes after everyone finished feedback.